src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { peopleFromServer } from '../data/people';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+type Props = {
+  onSelected?: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isListVisible, setList] = useState<boolean>(false);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [delay]);
```

Comment:
The `debounce` function should be wrapped in a `useCallback` to ensure that it does not get recreated on every render. This is important for performance optimization. Consider using `useCallback` to wrap the `debounce` function.